### PR TITLE
Log end of test phases and broker host setup and teardown

### LINKED ISSUE #12438: RFE: Create a log record when we go for host checkin
## Problem statement
It is difficult for the user to distinguish where the host check-in started in the log file - what log records are being logged from the check-in phase.

## Solution
Call the logger once we start checking in and log that the check-in process started.

### PR DESCRIPTION
**Pull request by ogajduse**
## Problem statement
* There is no way to distinguish between the start and end of individual test phases (setup, call, teardown) in our logs
* There is no feasible way to tell which test failed if the test suite is still running
* There is no contextual message informing the log reader about broker host `setup` and `teardown` being executed

## Solution
* Use the `pytest_runtest_logreport` hook to log the result of each phase.
* Add logging to setup and teardown to improve context awareness of our logs - fixes #12438.

## RoI
This functionality increases context awareness for those who read logs.

## Notes
* pytest documentation: https://docs.pytest.org/en/latest/reference/reference.html#pytest.hookspec.pytest_runtest_logreport

### CODE REVIEW

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user_by_ip[rhel9] tests/foreman/cli/test_remoteexecution.py::TestAnsibleREX::test_positive_run_effective_user_job[rhel9] tests/foreman/cli/test_host.py::test_positive_erratum_applicability[rhel9] tests/foreman/destructive/test_packages.py::test_positive_all_packages_update

**Comment by ogajduse** [comment]
Note: Whoever merges the PR, needs to squash the commits.

File: robottelo/hosts.py [hunk 1]
```diff
@@ -373,15 +373,21 @@ def clean_cached_properties(self):
                 del self.__dict__[name]
 
     def setup(self):
+        logger.debug('START: setting up host %s', self)
```
[the remark this answers is not in the document]
Comment (ogajduse) [reply robottelo/hosts.py]: addressed

**Review by Gauravtalreja1 — APPROVED** [review]

**Review by jameerpathan111 — APPROVED** [review]

File: robottelo/hosts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ogajduse) [reply robottelo/hosts.py:376]: I have removed the patch as the host type is already contained in `self`.

**Review by JacobCallahan — APPROVED** [review]

**Review by rplevka — APPROVED** [review]

**Pull request merged**